src/modules/claims/claims.service.ts

```diff
@@ -1403,22 +1408,124 @@ export class ClaimsService {
     );
   }
 
+  /**
+   * Verifies:
+   * - That credential proof is valid
+   * - That credential was issued by authorized issuer
+   *
+   * @param {VerifiableCredential<RoleCredentialSubject} vc to be verified
+   * @return Boolean indicating if verified and array of error messages
+   */
+  async verifyVc(
+    vc: VerifiableCredential<RoleCredentialSubject>
+  ): Promise<CredentialVerificationResult> {
+    const errors: string[] = [];
+    const issuerDID = this._signerService.did;
+    const proofVerified = await this._verifiableCredentialService.verify(vc);
+    if (!proofVerified) {
+      errors.push(ERROR_MESSAGES.PROOF_NOT_VERIFIED);
+    }
+    const role = vc.credentialSubject.role.namespace;
+    let issuerVerified = true;
+    try {
+      await this._vcIssuerVerifier.verifyIssuer(issuerDID, role);
+    } catch (e) {
+      issuerVerified = false;
+      errors.push((e as Error).message);
+    }
+    return {
+      errors,
+      isVerified: proofVerified && issuerVerified,
+    };
+  }
+
+  /**
+   * Verifies:
+   * - That off-chain claim was issued by authorized issuer
+   * - That off-chain claim proof is valid
+   *
+   * @param subjectDID The DID to try to resolve a credential for
+   * @param roleNamesapce The role to try to get a credential for. Should be a full role namespace (for example, "myrole.roles.myorg.auth.ewc")
+   * @return Boolean indicating if verified and array of error messages
+   */
+  async verifyOffChainClaim(
+    subjectDID: string,
+    roleNamespace: string
+  ): Promise<CredentialVerificationResult> {
+    const errors: string[] = [];
+    const issuerDID = this._signerService.did;
+    const claimIssuerVerifier = new ClaimIssuerVerification(
+      this._signerService.provider,
+      this._didRegistry.registrySettings,
+      this._credentialResolver,
+      this._issuerResolver
+    );
+      const issuerVerified = await claimIssuerVerifier.verifyIssuer(
+        issuerDID,
+        roleNamespace
+      );
+
+    if (!issuerVerified) {
+      errors.push(ERROR_MESSAGES.OFFCHAIN_ISSUER_NOT_AUTHORIZED);
+    }
+    let proofVerified = true;
+    try {
+      await claimIssuerVerifier.verifyIssuance(subjectDID, roleNamespace);
+    } catch (e) {
+      proofVerified = false;
+      errors.push((e as Error).message);
+    }
+    return {
+      errors: errors,
+      isVerified: proofVerified && issuerVerified,
+    };
+  }
+
+  /**
+   * Resolve a credential from storage and verify its proof/signature and its issuer's authority
+   *
+   * @param subjectDID The DID to try to resolve a credential for
+   * @param roleNamesapce The role to try to get a credential for. Should be a full role namespace (for example, "myrole.roles.myorg.auth.ewc")
+   * @return void. Returns "Proof Not Verified" error if VC not verified. Returns error if issuer not verified
+   */
+  async resolveCredentialAndVerify(
```

Comment:
@JGiter @jrhender This is the new verify code based on previous comments. I have a few of the tests passing, but questions on others - I think maybe I am misunderstanding the limitations of ew-credentials, but I don't think I need to mock anything else out as _verifyIssuer is verifying the credential successfully.